### PR TITLE
Update to .NET 5, update build metadata, include VS Code tasks

### PR DESCRIPTION
**Pull request by JohnathonMohr**
- Update CLI and test projects to use .NET 5
- Rename CLI output to TemplateAnalyzer (.exe and .dll)
- Add a `<Description>` field to CLI and Core projects
- Add common properties to new Build.Directory.props file
- Add VS Code configuration files for building and running
- Update documentation for contributing

### CODE REVIEW

File: src/Analyzer.Cli/Analyzer.Cli.csproj [hunk 1]
```diff
@@ -2,13 +2,15 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net5.0</TargetFramework>
+    <AssemblyName>TemplateAnalyzer</AssemblyName>
+    <Description>A command line interface for Microsoft.Azure.Templates.Analyzer.Core - an ARM Template scanner for security misconfiguration and best practices</Description>
```
Comment (VeraBE) [inline src/Analyzer.Cli/Analyzer.Cli.csproj:7]: "Misconfigurations" and template with lowercase?

File: src/Analyzer.Core/Analyzer.Core.csproj [hunk 2]
```diff
@@ -4,6 +4,7 @@
     <TargetFramework>netstandard2.1</TargetFramework>
     <RootNamespace>Microsoft.Azure.Templates.Analyzer</RootNamespace>
     <AssemblyName>Microsoft.Azure.Templates.Analyzer.Core</AssemblyName>
+    <Description>An ARM Template scanner for security misconfiguration and best practices</Description>
```
Comment (VeraBE) [inline src/Analyzer.Core/Analyzer.Core.csproj:7]: Same here

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": ".NET Core Launch (console)",
+            "type": "coreclr",
+            "request": "launch",
+            "preLaunchTask": "build",
+            "program": "${workspaceFolder}/src/Analyzer.Cli/bin/Debug/net5.0/TemplateAnalyzer.dll",
+
+            // Runs the CLI against the currently open ARM template file.
```
Comment (VeraBE) [inline .vscode/launch.json:11]: You mean the currently open template in VSCode?

**Review by VeraBE — COMMENTED** [review]
Looks good, just small comments

**Review by VeraBE — APPROVED** [review]

File: .vscode/launch.json (hunk 3 above)
Comment (JohnathonMohr) [reply .vscode/launch.json:11]: Yes - I'll update to be more clear.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,14 +1,21 @@
...
+* The .NET solution can be built with the `dotnet build` command.
  
 ## Developing
- 
+
+### Environment
+* [Visual Studio Code](https://code.visualstudio.com/) with the [C# extension](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csharp) is a great way to get started.
+  * Start VS Code and open the root directory where the code is cloned locally (File->Open Folder...).
+* Run the `build` task (Terminal->Run Task...->build) in VS Code to build the Template BPA.
+* Try analyzing a template: open a template file in VS Code and then run the `.NET Core Launch (console)` launch configuration (Run->Start Debugging).
```
Comment (yane3628) [inline CONTRIBUTING.md:16]: My VS Code environment is missing the coreclr Extension. Can you add this as a prereq for using VSCode? How do you find this in the Extension Marketplace?

**Review by yane3628 — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 4 above)
Comment (VeraBE) [reply CONTRIBUTING.md:16]: I installed VSCode with only the C# extension and was able to follow the instructions :O

**Review by yane3628 — APPROVED** [review]

**Pull request merged**